### PR TITLE
[WIP] breaking(): Deprecate req.body.data in favor of req.body

### PR DESCRIPTION
**Pull request by XavierGeerinck**
Signed-off-by: Xavier Geerinck <xavier.geerinck@gmail.com>

# Description

We deprecated `req.body.data` in favor of `req.body` on the HTTP Pub Sub component. This means that data will now be in `req.body.data` and users should fetch accordingly.

The advantage of working this way is that providers that don't return data can be parsed and users can as well get the headers around it (e.g., source, pubsubname, datacontenttype, ...)

More practical, instead of returning:

```javascript
{
    hello: "world"
}
```

We will be returning:

```javascript
{
  data: { hello: 'world' },
  datacontenttype: 'application/json',
  id: '80e37aaa-a7e4-4484-a74e-2fd9e0194c47',
  pubsubname: 'pubsub-redis',
  source: 'test-suite',
  specversion: '1.0',
  time: '2022-11-29T11:30:16+01:00',
  topic: 'topic-1',
  traceid: '00-d0a8afe02330296da9046a7d827eecbe-8876a7ac21b31b6f-01',
  traceparent: '00-d0a8afe02330296da9046a7d827eecbe-8876a7ac21b31b6f-01',
  tracestate: '',
  type: 'com.dapr.event.sent'
}
```

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

- [x] Code compiles correctly
- [x] Created/updated tests
- [ ] Extended the documentation

### CODE REVIEW

File: src/implementation/Server/HTTPServer/HTTPServerImpl.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ export default class HTTPServerImpl {
         // Parse the data of the body, we prioritize fetching the data key in body if possible
         // i.e. Redis returns { data: {} } and other services return {}
         // @todo: This will be deprecated in an upcoming major version and only req.body will be returned
```
Comment (shubham1172) [inline src/implementation/Server/HTTPServer/HTTPServerImpl.ts:90]:
```suggestion
```

Comment (shubham1172) [reply src/implementation/Server/HTTPServer/HTTPServerImpl.ts:90]: We should also fix the comment above.

File: test/e2e/http/server.test.ts [hunk 2]
```diff
@@ -112,7 +112,6 @@ describe("http/server", () => {
 
       // Delay a bit for event to arrive
       await new Promise((resolve, _reject) => setTimeout(resolve, 250));
-      expect(mockBindingReceive.mock.calls.length).toBe(1);
```
Comment (shubham1172) [inline test/e2e/http/server.test.ts:115]: Why are we removing these?

File: test/e2e/http/server.test.ts [hunk 3]
```diff
@@ -239,6 +235,29 @@ describe("http/server", () => {
       expect(res).toEqual(true);
     });
 
+    it("should receive the entire body instead of just data, this also contains the content type, time, traces, ...", async () => {
```
Comment (shubham1172) [inline test/e2e/http/server.test.ts:238]:
```suggestion
    it("should be able to receive the entire cloud event", async () => {
```

**Review by shubham1172 — CHANGES_REQUESTED** [review]
Some small changes!

File: test/e2e/http/server.test.ts (hunk 2 above)
Comment (XavierGeerinck) [reply test/e2e/http/server.test.ts:115]: Re-added

**Comment by XavierGeerinck** [comment]
Closing in favor of #443

**Pull request closed without merging**